### PR TITLE
Remove unused route

### PR DESCRIPTION
**Pull request by Gusted**
- Couldn't find any reference of this route being used.

### CODE REVIEW

File: routers/web/repo/pull.go [hunk 1]
```diff
@@ -1208,44 +1207,6 @@ func CompareAndPullRequestPost(ctx *context.Context) {
...
-			ctx.Error(http.StatusNotFound)
-		} else {
-			ctx.ServerError("GetUserByID", err)
-		}
-		return
-	}
-
-	log.Trace("TriggerTask '%s/%s' by %s", repo.Name, branch, pusher.Name)
-
-	go pull_service.AddTestPullRequestTask(pusher, repo.ID, branch, true, "", "")
```
Comment (6543) [inline routers/web/repo/pull.go:1245]: this func if run can repair broken pulls if PullRequestTestTask was messed due to ... a restart, misconfigured pull hooks ...

so i would call this func "repair pull" - but no idea if we like to have such func in the web interface

**Comment by 6543** [comment]
I quote @zeripath: "... only destroy things if you know what they where for in the past ..." (or similar)

**Comment by Gusted** [comment]
> I quote @zeripath: "... only destroy things if you know what they where for in the past ..." (or similar)

I took my detective hat and put it on my head and started a investigation journey.

- The last time the route was modified was in #1290
- A lot of the code comes from gogs https://github.com/go-gitea/gitea/commit/3362b3a44f9f4e0e482b08151e298f7809eefc59 https://github.com/go-gitea/gitea/commit/3f7f4852efaaa56a0dada832dc652a1fc8869ae7 https://github.com/go-gitea/gitea/commit/0fbb8c8826771e92e890fb1c72b356d3e62c7b01
- In one of the comments you can see where it was used https://github.com/go-gitea/gitea/commit/3362b3a44f9f4e0e482b08151e298f7809eefc59#diff-fb0c696bc104f613e5b601c77c2eba916dd636d5948c9b5c8e85ea92ee5b07e6R113 (Which is no longer in the current state of code)

- It was removed by lunny in https://github.com/go-gitea/gitea/commit/1773e88643a2df7e7efbe86ec424409b45a4d576

File: routers/web/repo/pull.go (hunk 1 above)
Comment (Gusted) [reply routers/web/repo/pull.go:1245]: > this func if run can repair broken pulls if PullRequestTestTask was messed due to ... a restart, misconfigured pull hooks ...
> 
> so i would call this func "repair pull" - but no idea if we like to have such func in the web interface

Their are some other references to this function(In `Merge`, `Update` `pushUpdates`) So it's already used where it should be.

**Review by 6543 — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
:robot:

**Pull request merged**